src/nunit.analyzers/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzer.cs

```diff
@@ -53,47 +83,235 @@ private static void AnalyzeAttribute(SyntaxNodeAnalysisContext context)
+                            sourceTypeNotIEnumerableDescriptor,
+                            attributeNode.ArgumentList.Arguments[0].GetLocation(),
+                            sourceType.Name));
+                    }
+                    else if (!typeHasDefaultConstructor)
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(
+                            sourceTypeNoDefaultConstructorDescriptor,
+                            attributeNode.ArgumentList.Arguments[0].GetLocation(),
+                            sourceType.Name));
+                    }
+
+                    return;
+                }
+
+                var symbol = GetMember(context, attributeInfo);
+                if (symbol is null)
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(
+                       missingSourceDescriptor,
+                       syntaxNode.GetLocation(),
+                       stringConstant));
+                }
+                else
+                {
+                    if (attributeInfo.IsStringLiteral)
                     {
-                        var stringConstant = literal.Token.ValueText;
                         context.ReportDiagnostic(Diagnostic.Create(
                             considerNameOfDescriptor,
-                            literal.GetLocation(),
+                            syntaxNode.GetLocation(),
                             stringConstant));
                     }
-                    else
+
+                    if (!symbol.IsStatic)
                     {
-                        var literalText = literal.Token.ValueText;
-                        context.ReportDiagnostic(Diagnostic.Create(missingSourceDescriptor, literal.GetLocation(), literalText));
+                        context.ReportDiagnostic(Diagnostic.Create(
+                            sourceNotStaticDescriptor,
+                            syntaxNode.GetLocation(),
+                            stringConstant));
+                    }
+
+                    switch (symbol)
+                    {
+                        case IPropertySymbol property:
+                            ReportIfSymbolNotIEnumerable(property.Type);
+                            ReportIfParametersSupplied(attributeInfo.NumberOfMethodParameters);
+                            break;
+                        case IFieldSymbol field:
+                            ReportIfSymbolNotIEnumerable(field.Type);
+                            ReportIfParametersSupplied(attributeInfo.NumberOfMethodParameters);
+                            break;
+                        case IMethodSymbol method:
+                            ReportIfSymbolNotIEnumerable(method.ReturnType);
+
+                            if (method.Parameters.Length != attributeInfo.NumberOfMethodParameters)
+                            {
+                                // TODO Report mismatch of parameters

```

Comment:
That's pending that another issue, that you mentioned, right?

Reply:
Yes. All the TODOs will be solved by #214. I just wanted to do it in steps.